File: app/src/main/java/de/cineaste/android/adapter/NearbyUserAdapter.java
package de.cineaste.android.adapter;

import android.content.Context;
import android.content.res.Resources;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import de.cineaste.android.R;
import de.cineaste.android.entity.NearbyMessage;

public class NearbyUserAdapter extends RecyclerView.Adapter<NearbyUserAdapter.ViewHolder> {

    private final List<NearbyMessage> nearbyMessages;
    private final int rowLayout;
    private final Context context;

    public NearbyUserAdapter(
            List<NearbyMessage> nearbyMessages,
            Context context ) {
        this.nearbyMessages = nearbyMessages;
        this.rowLayout = R.layout.card_nearby_user;
        this.context = context;
    }

    @Override
    public int getItemCount() {
        return nearbyMessages == null ? 0 : nearbyMessages.size();
    }

    @Override
    public ViewHolder onCreateViewHolder( ViewGroup parent, int viewType ) {
        View v = LayoutInflater.from( parent.getContext() ).inflate( rowLayout, parent, false );
        return new ViewHolder( v );
    }

    @Override
    public void onBindViewHolder( final ViewHolder holder, int position ) {
        final NearbyMessage nearbyMessage = nearbyMessages.get( position );
        holder.userName.setText( nearbyMessage.getUserName() );
        Resources resources = context.getResources();
        int count = nearbyMessage.getMovies().size();
        holder.movieCounter.setText( resources.getQuantityString( R.plurals.movieCounter, count, count ) );
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private final TextView userName, movieCounter;

        public ViewHolder( final View itemView ) {
            super( itemView );
            userName = itemView.findViewById( R.id.userName_tv );
            movieCounter = itemView.findViewById( R.id.movie_counter_tv );
        }
    }
}

File: app/src/main/kotlin/de/cineaste/android/adapter/NearbyUserAdapter.kt
package de.cineaste.android.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import de.cineaste.android.R
import de.cineaste.android.entity.movie.NearbyMessage
import de.cineaste.android.listener.UserClickListener

class NearbyUserAdapter(
    private val nearbyMessages: List<NearbyMessage> = listOf(),
    private val context: Context,
    private val listener: UserClickListener
) : RecyclerView.Adapter<NearbyUserAdapter.ViewHolder>() {
    private val rowLayout: Int = R.layout.card_nearby_user

    override fun getItemCount(): Int {
        return nearbyMessages.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(rowLayout, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val nearbyMessage = nearbyMessages[position]
        holder.assignData(nearbyMessage)
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        private val userName: TextView = itemView.findViewById(R.id.userName_tv)
        private val movieCounter: TextView = itemView.findViewById(R.id.movie_counter_tv)

        fun assignData(nearbyMessage: NearbyMessage) {
            userName.text = nearbyMessage.userName
            val resources = context.resources
            val count = nearbyMessage.movies.size
            movieCounter.text = resources.getQuantityString(R.plurals.movieCounter, count, count)
            itemView.setOnClickListener {
                listener.onUserClickListener(nearbyMessage)
            }
        }

        override fun onClick(v: View?) {
            listener.onUserClickListener(nearbyMessages[adapterPosition])
        }
    }
}
